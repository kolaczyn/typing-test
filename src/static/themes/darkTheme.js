import { DARK_THEME } from '../../reducers/theme/actions';

const darkTheme = {
  palette: {
    box: '#222',
    border: '#1c1e21',
    background: '#303030',
    text: '#fff',

    green: '#55efc4',
    blue: '#00a1e4',
    red: '#e17055',
    yellow: '#e8e9a1',
    lightGray: 'bisque',
  },
  themeName: DARK_THEME,
};

export default darkTheme;
