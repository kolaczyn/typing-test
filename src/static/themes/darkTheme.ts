import Action from '../../reducers/theme/actions';
import ThemeType from './ThemeType';

// https://flatuicolors.com/palette/defo
const darkTheme: ThemeType = {
  palette: {
    box: '#2c3e50',
    border: '#27394B',
    background: '#34495e',
    text: '#ecf0f1',

    green: '#2ecc71',
    blue: '#3498db',
    red: '#f96454',
    yellow: '#f1c40f',
    lightGray: '#bdc3c7',
    darkGray: '#bdc3c7',
  },
  themeName: Action.DARK_THEME,
};

export default darkTheme;
