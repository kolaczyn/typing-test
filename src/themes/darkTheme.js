export default {
  palette: {
    box: '#222',
    background: '#303030',
    lightGray: 'bisque',
    green: '#55efc4',
    red: '#e17055',
    text: '#fff',
  },
  containerWidth: '768px',
};
