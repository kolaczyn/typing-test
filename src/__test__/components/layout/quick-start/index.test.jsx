import React from 'react';
import { ThemeProvider } from 'styled-components';
import { cleanup, render, fireEvent } from '@testing-library/react';

import QuickStart from '../../../../components/layout/quick-start';
import darkTheme from '../../../../static/themes/darkTheme';

afterEach(cleanup);

describe('<QuickStart />', () => {
  it('renders <QuickStart />', () => {
    const { container } = render(
      <ThemeProvider theme={darkTheme}>
        <QuickStart />
      </ThemeProvider>,
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  it('checks if closing works correctly', () => {
    const { container, getByTestId } = render(
      <ThemeProvider theme={darkTheme}>
        <QuickStart />
      </ThemeProvider>,
    );

    expect(getByTestId('close-button')).toBeTruthy();
    fireEvent.click(getByTestId('close-button'));
    expect(container.firstChild).toMatchSnapshot();
  });
});
