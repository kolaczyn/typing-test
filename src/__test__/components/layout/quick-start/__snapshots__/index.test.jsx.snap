// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<QuickStart /> checks if closing works correctly 1`] = `null`;

exports[`<QuickStart /> renders <QuickStart /> 1`] = `
<section
  class="styles__Container-sc-9qjjy7-0 dsJXrG styles-oh4xco-0 eyiePF"
  title="11"
>
  <header
    class="styles__Header-sc-9qjjy7-1 ejqAhC"
  >
    Quick Start
  </header>
  <button
    class="button-sc-16cxp6v-0 styles-w4qo46-0 bGmkNd fZzkFu"
    data-testid="close-button"
  >
    <span
      class="material-icons"
    >
      clear
    </span>
  </button>
  Welcome! Lorem ipsum dolor sit amet consectetur adipisicing elit. Expedita provident reprehenderit maiores non, quisquam et laudantium beatae eaque dicta, consectetur impedit explicabo fuga autem pariatur distinctio, placeat est?
</section>
`;
